daemon/container_operations.go

```diff
@@ -980,6 +980,9 @@ func (daemon *Daemon) ConnectToNetwork(container *container.Container, idOrName
 	if endpointConfig == nil {
 		endpointConfig = &networktypes.EndpointSettings{}
 	}
+	container.Lock()
+	defer container.Unlock()
```

Comment:
We are locking around a big chunk of code that includes calls which doesn't need the container lock (for ex: daemon.FindNetwork(). Especially in the case the container is running `connectToNetwork` does lot of things. If possible the lock should be more fine-grained, only when modifying the container object.